paddle/operators/sequence_conv_op.cc

```diff
@@ -0,0 +1,165 @@
+      PADDLE_ENFORCE(
+          padding_dim[0] == total_pad && padding_dim[1] == input_width,
+          "Input(PaddingData)'s shape is not consistent with 'context_start' "
+          "and 'context_length'.");
+    }
+
+    in_dims[1] = 1;
+    ctx->SetOutputDim("Out", in_dims);
+  }
+};
+
+class SequenceConvGradOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+ protected:
+  void InferShape(framework::InferShapeContext* ctx) const override {
+    PADDLE_ENFORCE(ctx->HasInput(framework::GradVarName("Out")),
+                   "Gradient of output(Out) should not be null.");
+    PADDLE_ENFORCE(ctx->HasInput("X"), "The input(X) should not be null.");
+
+    if (ctx->Attrs().Get<bool>("padding_trainable") &&
+        ctx->HasOutput(framework::GradVarName("PaddingData"))) {
+      ctx->SetOutputDim(framework::GradVarName("PaddingData"),
+                        ctx->GetInputDim("PaddingData"));
+    }
+    if (ctx->HasOutput(framework::GradVarName("X"))) {
+      ctx->SetOutputDim(framework::GradVarName("X"), ctx->GetInputDim("X"));
+    }
+    if (ctx->HasOutput(framework::GradVarName("Filter"))) {
+      ctx->SetOutputDim(framework::GradVarName("Filter"),
+                        ctx->GetInputDim("Filter"));
+    }
+  }
+};
+
+class SequenceConvOpMaker : public framework::OpProtoAndCheckerMaker {
+ public:
+  SequenceConvOpMaker(framework::OpProto* proto,
+                      framework::OpAttrChecker* op_checker)
+      : OpProtoAndCheckerMaker(proto, op_checker) {
+    AddInput("X",
+             "(A float LoDTensor) the input of SequenceConvOp, a vector of "
+             "2-D matrix of size (minibatch, number_of_input_features).");
+    AddInput("PaddingData",
+             "(Tensor) the input of SequenceConvOp, a vector of "
+             "2-D matrix of size (up_pad + down_pad, "
+             "number_of_input_features). ")
+        .AsDispensable();
+    AddInput("Filter",
+             "(Tensor) the input of SequenceConvOp, a vector of "
+             "2-D matrix of size (context_length x number_of_input_features).");
+    AddOutput("Out",
+              "(A float LoDTensor) the output of SequenceConvOp, a vector "
+              "of 2-D matrix of size (minibatch, 1).");
+
+    AddAttr<bool>("padding_trainable",
+                  "(bool, default false) the padding data of SequenceConvOp "
+                  "is trainable or not.")
+        .SetDefault(false);
+    AddAttr<int>("context_length",
+                 "(int, default 3) the context_length of SequenceConvOp.")
+        .SetDefault(3)
+        .GreaterThan(0);
+    AddAttr<int>("context_start",
+                 "(int, default 0) the context_start of SequenceConvOp.")
+        .SetDefault(0);
+    AddAttr<int>("context_stride",
+                 "(int, default 1) the context_stride of SequenceConvOp. "
+                 "Currently, sequence_project_op only support "
+                 "context_stride=1.")
```

Comment:
The explanation of  `context_start` and `context_stride` is also not clearly.

Reply:
Done